### ISSUE TITLE
Collection Deadlock

### ISSUE THREAD

**Issue by adamretter** [eXist-db/exist#1705]
Can be demonstrated by running `RESTServiceTest.java` in parallel, see: https://github.com/eXist-db/exist/blob/develop/test/src/org/exist/http/RESTServiceTest.java#L65

**Comment by adamretter** [comment]
@duncdrum Yes, it is also possible to make happen with `EvalTest.java`. However, if we do that, then the test suite never completes because of the deadlock.

**Comment by duncdrum** [comment]
well at least the testsuite would be working ;-)
